bin/cliain/src/keys.rs

```diff
@@ -1,31 +1,31 @@
 use aleph_client::{
-    rotate_keys as rotate, rotate_keys_raw_result, set_keys as set, staking_bond, Connection,
-    SessionKeys,
+    rotate_keys as rotate, rotate_keys_raw_result, set_keys as set, staking_bond, AnyConnection,
+    RootConnection, SessionKeys, SignedConnection,
 };
 use log::info;
 use primitives::staking::MIN_VALIDATOR_BOND;
 use substrate_api_client::{AccountId, XtStatus};
 
-pub fn prepare_keys(connection: Connection, controller_account_id: AccountId) {
+pub fn prepare_keys(connection: RootConnection, controller_account_id: AccountId) {
     staking_bond(
-        &connection,
+        &connection.as_signed(),
```

Comment:
I understand now why `staking_bond` takes `RootConnection` , can we somehow cast `RootConnection ` to signed connections here?

Reply:
Nvm I was under the impression this code won't compile but it will.